swarm/network/kademlia_test.go

```diff
@@ -89,65 +94,209 @@ func TestNeighbourhoodDepth(t *testing.T) {
+		t.Fatal(err)
+	}
+
+	// know one peer and connected
+	// healthy
+	On(k, "11100000")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// know two peers, only one connected
+	// unhealthy
+	Register(k, "11111100")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// know two peers and connected to both
+	// healthy
+	On(k, "11111100")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// know three peers, connected to the two deepest
+	// healthy
+	Register(k, "00000000")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// know three peers, connected to all three
+	// healthy
+	On(k, "00000000")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// add fourth peer deeper than current depth
+	// unhealthy
+	Register(k, "11110000")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// connected to three deepest peers
+	// healthy
+	On(k, "11110000")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+
+	// add additional peer in same bin as deepest peer
+	// unhealthy
+	Register(k, "11111101")
+	log.Trace(k.String())
+	if err := assertHealthSimple(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+	// four deepest of five peers connected
+	// healthy
+	On(k, "11111101")
+	if err := assertHealthSimple(t, k, true); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestHealthPotential(t *testing.T) {
+	k := newTestKademlia("11111111")
+	if err := assertHealthPotential(t, k, false); err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+// retrieves the health object based on the current connectivity of the given kademlia
+func getHealth(k *Kademlia) *Health {
+	kid := common.Bytes2Hex(k.BaseAddr())
+	addrs := [][]byte{k.BaseAddr()}
+	k.EachAddr(nil, 255, func(addr *BzzAddr, po int, _ bool) bool {
+		addrs = append(addrs, addr.Address())
+		return true
+	})
+	pp := NewPeerPotMap(k.MinProxBinSize, addrs)
+	return k.Healthy(pp[kid])
+}
+
+// evaluates the simplest definition of health:
+// all conditions must be true:
+// - we at least know one peer
+// - we know all neighbors
+// - we are connected to all known neighbors
+func assertHealthSimple(t *testing.T, k *Kademlia, expectHealthy bool) error {
+	healthParams := getHealth(k)
+	health := healthParams.KnowNN && healthParams.GotNN && healthParams.CountKnowNN > 0
+	if expectHealthy != health {
+		return fmt.Errorf("expected kademlia health %v, is %v\n%v", expectHealthy, health, k.String())
 	}
+	return nil
+}
+
+// evaluates healthiness by taking into account potential connections
+// additional conditions for healthiness
+// - IF we know of peers in bins shallower than depth, connected to at least HealthBinSize of them
+func assertHealthPotential(t *testing.T, k *Kademlia, expectHealthyAndPotent bool) error {
```

Comment:
`testing.T` is not used in this function. If you just call `t.Fatal` instead of returning the `error`, this assert function will be more compact and easier to use in tests.